Blocky utility 01 blocky logger

kamphoff opened the pull request:
maakt geen gebruik van de cmake structuur
heeft nog niet G test (structuur), wel een test class, met een soort example

kamphoff committed:
added logging

Deze feature kan:
- Errors printen
- Warnings printen
- Info printen
- Debugs printen

kamphoff committed:
cmake nog steets niet ok, maar voeg later toe als Lars de juist versie op dev heeft staan

kamphoff committed:
added macro

change to LogLevel to string string naming

macro's are supported

kamphoff committed:
added function name to log

see funcSignToString if change to include class name is needed.

kamphoff committed:
change to use define for file location

kamphoff committed:
change ERR to ERROR since alignment is the same for DEBUG 5 characters

kamphoff committed:
[fix] F-BU-01-C: Loggen in formaat

use of class name instead of function calling function is declared in a class

kamphoff committed:
[fix] create oneTestClass to call Test

kamphoff committed:
[fix] folowing style guide

kamphoff committed:
[fix]: changed test file

kamphoff committed:
[fix] style guide

Dogukan-lab commented on CMakeLists.txt:
Vermijd in elk geval de .cpp en .hpp direct in de main cmake te stoppen

Dogukan-lab commented on engine/logging/BLogger.cpp:
```diff
@@ -0,0 +1,104 @@
+#include <iomanip>
+#include <iostream>
+
+using namespace std::chrono;
```
Geen using namespace!

Dogukan-lab commented on engine/logging/BLogger.cpp:
```diff
@@ -0,0 +1,104 @@
+
+using namespace std::chrono;
+
+BLogger::BLogger(const std::string &filename) {
```
Wat als we alleen naar de console willen loggen?

Dogukan-lab commented on engine/logging/BLogger.cpp:
```diff
@@ -0,0 +1,104 @@
+
+
+void BLogger::Log(const LogLevel level, const std::string &funcName, const std::string &message) {
+	auto logMessage = _MakeTimeStamp();
```
_PascalCase is blijkbaar niet allowed. Maak van je private functies _camelCase

Dogukan-lab commented on engine/logging/BLogger.hpp:
```diff
@@ -0,0 +1,49 @@
+
+#define LOG_TO_CONSOLE 1
+#define MAX_FUNCTION_NAME_LENGHT 15
+#define LOG_FILE "logfile.txt"
```
Deze wegdoen voor juist een standaard value binnen de constructor

Dogukan-lab commented on engine/logging/BLogger.hpp:
```diff
@@ -0,0 +1,49 @@
+#include <fstream>
+
+#define LOG_TO_CONSOLE 1
+#define MAX_FUNCTION_NAME_LENGHT 15
```
Maak van beide defines constexpr, want dit is C-style programmeren.

Dogukan-lab commented on engine/logging/BLogger.hpp:
```diff
@@ -0,0 +1,49 @@
+        void _WriteLog(const std::stringstream &logMessage);
+};
+
+inline BLogger bLogger(LOG_FILE);
```
Waarvoor is dit exact?

kamphoff replied:
Clion: Definition of global variable 'BLogger bLogger' in a header file should have an 'inline' specifier. Maar kan ook static zijn, is er voor, zodat bLogger niet door user hoeft worden ge init, en zodat er ook niet meerdere keren de init zou kunnen worden aangeroepen

Dogukan-lab commented on engine/testing/BLoggerTest.cpp:
```diff
@@ -0,0 +1,42 @@
+		BLOCKY_ENGINE_INFO("Program update: " + std::to_string(i));
+
+		//delay
+		for (int j = 0; j < 100000000; j++) { if (i == j) { j = i; } }
```
Simpelweg, nee haal dat er of uit, of gebruik iets zoals de chrono time library om een timer te maken voor een delay

Dogukan-lab reviewed (changes requested):
Overall well ongeveer wat in gedachten had, verder is het heel belangrijk dat je op de hoogte bent van het feit dat de tests dmv GoogleTests opnieuw gemaakt zullen worden. Zie opmerking voor de changes

kamphoff committed:
[fix]: pull request 2

fixed:
- use of CMake
- c style code
- incorrect styling

kamphoff committed:
[fix]: REMOVE_FUNCTION & RETURN_TYPE

fixed flowing that did not behave as name suggests:
- REMOVE_FUNCTION_NAME_FROM_CLASSES
- REMOVE_RETURN_TYPE

Dogukan-lab reviewed (commented):
Overall zie ik idd verbetering in de code! Ik wil wel dat 1 iemand nog dit reviewed voordat het naar dev wordt gemerged.

LarsingDash reviewed (approved):
Ziet er goed uit. Loop in de test is best lelijk maar zal toch vervangen worden zodra de daadwerkelijke testing suite staat. Mijn enige vraag is: wat is het verschil tussen INFO en DEBUG als de logger toch enkel zichtbaar is in DEBUG mode? Overwegen die nog weg te halen, verder is t überhaupt approved
